pkg/apis/configuration/validation/validation_test.go

```diff
@@ -1658,16 +1689,64 @@ func TestValidateTime(t *testing.T) {
 	}
 }
 
+func TestValidateOffset(t *testing.T) {
+	var validInput = []string{"", "1", "10k", "11m", "1K", "100M", "5G"}
+	for _, test := range validInput {
+		allErrs := validateOffset(test, field.NewPath("offset-field"))
+		if len(allErrs) != 0 {
+			t.Errorf("validateOffset(%q) returned an error for valid input", test)
+		}
+	}
+
+	var invalidInput = []string{"55mm", "2mG", "6kb", "-5k", "1L", "5Gb"}
+	for _, test := range invalidInput {
+		allErrs := validateOffset(test, field.NewPath("offset-field"))
+		if len(allErrs) == 0 {
+			t.Errorf("validateOffset(%q) didn't return error for invalid input.", test)
+		}
+	}
+}
+
+func TestValidateBuffer(t *testing.T) {
+	validbuff := &v1alpha1.UpstreamBuffers{Number: 8, Size: "8k"}
+	allErrs := validateBuffer(validbuff, field.NewPath("buffers-field"))
+
+	if len(allErrs) != 0 {
+		t.Errorf("validateBuffer returned errors %v valid input %v", allErrs, validbuff)
+	}
+
+	invalidbuff := []*v1alpha1.UpstreamBuffers{
+		{
```

Comment:
small comment about the input for test cases - please change the input parameters so that only 2 of the fields (size or number) is invalid. For example,

			Number: -8,
			Size:   "15m",

			Number: 8,
			Size:   "15G",

                         Size: 8,
			Size:  "",

Otherwise, the test won't  fail if the validation for one of the parameter is buggy.